banzai/data.py

```diff
@@ -39,14 +39,14 @@ def __init__(self, data: Union[np.array, Table], meta: Union[dict, fits.Header],
         else:
             self.data = self._init_array(data)
         self.meta = meta.copy()
-        self._validate_mask(mask)
+        self._validate_array(mask)
         self.mask = self._init_array(mask, dtype=np.uint8)
         self.name = name
 
-    def _validate_mask(self, mask):
-        if mask is not None:
-            if mask.shape != self.data.shape:
-                raise ValueError('Mask must have the same dimensions as the data')
+    def _validate_array(self, array_data):
```

Comment:
Renamed this from `_validate_mask` to `_validate_array` since we're now using it to validate the size of the uncertainty map and bad pixel masks